__test__/unit/adapters/collateralized_simple_interest_loan_adapter.spec.ts

```diff
@@ -239,3 +254,364 @@ describe("Collateralized Terms Contract Interface (Unit Tests)", () => {
+                            scenario_3.fullLoanOrder,
+                        ),
+                    ).resolves.toEqual(scenario_3.debtOrder);
+                });
+            });
+        });
+    });
+
+    describe("#fromDebtOrder()", () => {
+        describe("argument does not conform to the DebtOrderWithTermsSpecified schema", () => {
+            describe("malformed terms contract", () => {
+                test("should throw DOES_NOT_CONFORM_TO_SCHEMA", async () => {
+                    await expect(
+                        collateralizedSimpleInterestLoanAdapter.fromDebtOrder({
+                            ...scenario_1.debtOrder,
+                            termsContract: "invalid terms contract",
+                        }),
+                    ).rejects.toThrow("instance.termsContract does not match pattern");
+                });
+            });
+
+            describe("missing termsContract", () => {
+                test("should throw DOES_NOT_CONFORM_TO_SCHEMA", async () => {
+                    await expect(
+                        collateralizedSimpleInterestLoanAdapter.fromDebtOrder({
+                            ...scenario_1.debtOrder,
+                            termsContract: undefined,
+                        }),
+                    ).rejects.toThrow('instance requires property "termsContract"');
+                });
+            });
+
+            describe("missing termsContractParameters", () => {
+                test("should throw DOES_NOT_CONFORM_TO_SCHEMA", async () => {
+                    await expect(
+                        collateralizedSimpleInterestLoanAdapter.fromDebtOrder({
+                            ...scenario_1.debtOrder,
+                            termsContractParameters: undefined,
+                        }),
+                    ).rejects.toThrow('instance requires property "termsContractParameters"');
+                });
+            });
+
+            describe("missing principalAmount", async () => {
+                test("should throw DOES_NOT_CONFORM_TO_SCHEMA", async () => {
+                    await expect(
+                        collateralizedSimpleInterestLoanAdapter.fromDebtOrder({
+                            ...scenario_1.debtOrder,
+                            principalAmount: undefined,
+                        }),
+                    ).rejects.toThrow('instance requires property "principalAmount"');
+                });
+            });
+
+            describe("missing principalToken", async () => {
+                test("should throw DOES_NOT_CONFORM_TO_SCHEMA", async () => {
+                    await expect(
+                        collateralizedSimpleInterestLoanAdapter.fromDebtOrder({
+                            ...scenario_1.debtOrder,
+                            principalToken: undefined,
+                        }),
+                    ).rejects.toThrow('instance requires property "principalToken"');
+                });
```

Comment:
Need tests for when parameters defined in terms contract parameters are invalid -- i.e. `amortizationUnitType = 255`, no token tracked at registry index defined by collateralization parameters, etc.

Reply:
Great call. I actually caught a bug because we were missing these tests...

Reply:
Unless I'm missing something, I think the test for to token being tracked at registry index X is still missing.

Reply:
Is the test added in this commit not what you were imagining?

https://github.com/dharmaprotocol/dharma.js/pull/52/commits/262967a658a918bdaf5407f2a175542ce8815c2f

Reply:
so it's good you added that test!  another test that ought to be added, though, is one for the scenario in which the token index specified in the terms contract parameters does not have any token associated with it in the token registry.

Reply:
well, this is in the context of `fromDebtOrder`, and the debt order is user supplied -- there's no guarantee that the terms contract parameters aren't just nonsense keyboard mash.

Reply:
Got it. Crazy users. Test added.

Reply:
🤗

Reply:
Great thread

Reply:
Who invited @graemecode to the party? Buzzkill

Reply:
Where's Brendan?